packages/gatsby/index.d.ts

```diff
@@ -1118,12 +1196,13 @@ export interface ReplaceComponentRendererArgs extends BrowserPluginArgs {
     path: string
     "*": string
     uri: string
-    location: object
-    navigate: Function
+    location: Location
+    navigate: NavigateFn
     children: undefined
     pageResources: object
     data: object
-    pageContext: object
+    pageContext: { id: string; [key: string]: unknown }
```

Comment:
I'm not sure if `id` will always be in context. Won't it depend on what you are passing to `createPage` `context`?